src/tests/src/libs/antares/study/thermal-price-definition/thermal-price-definition.cpp

```diff
@@ -0,0 +1,290 @@
+#define BOOST_TEST_MODULE "test thermal price definition"
+
+#define WIN32_LEAN_AND_MEAN
+
+#include <boost/test/included/unit_test.hpp>
+#include <yuni/io/file.h>
+#include <filesystem>
+#include <fstream>
+
+#include <study.h>
+#include <antares/exception/LoadingError.hpp>
+
+#include "../checks/checkApplication.h"
+#include "cluster_list.h"
+#include "container.h"
+
+#define SEP Yuni::IO::Separator
+
+using namespace std;
+using namespace Antares::Data;
+
+std::filesystem::path getFolder()
+{
+    return std::filesystem::temp_directory_path();
+}
+
+void createIniFile()
+{
+    std::filesystem::path folder = getFolder();
+
+    std::ofstream outfile;
+    outfile.open(folder / "list.ini", std::ofstream::out | std::ofstream::trunc);
+
+    outfile << "[area]" << std::endl;
+    outfile << "name = area" << std::endl;
+    outfile << "group = Lignite" << std::endl;
+    outfile << "unitcount = 4" << std::endl;
+    outfile << "nominalcapacity = 900.000000" << std::endl;
+    outfile << "min-stable-power = 400.000000" << std::endl;
+    outfile << "min-up-time = 24" << std::endl;
+    outfile << "min-down-time = 24" << std::endl;
+    outfile << "co2 = 2.000000" << std::endl;
+    outfile << "marginal-cost = 23.000000" << std::endl;
+    outfile << "fixed-cost = 1700.000000" << std::endl;
+    outfile << "startup-cost = 70000.000000" << std::endl;
+    outfile << "market-bid-cost = 35.000000" << std::endl;
+    outfile << "costgeneration = useCostTimeseries" << std::endl;
+    outfile << "efficiency = 36.00000" << std::endl;
+    outfile << "variableomcost = 12.120000" << std::endl;
+
+    outfile.close();
+}
+
+void createFuelCostFile(int size)
+{
+    std::filesystem::path folder = getFolder();
+
+    std::ofstream outfile;
+    outfile.open(folder / "fuelCost.txt", std::ofstream::out | std::ofstream::trunc);
+
+    for (int i = 0; i < size; i++)
+    {
+        outfile << 1 << std::endl;
+    }
+}
+
+void createCo2CostFile(int size)
+{
+    std::filesystem::path folder = getFolder();
+
+    std::ofstream outfile;
+    outfile.open(folder / "CO2Cost.txt", std::ofstream::out | std::ofstream::trunc);
+
+    for (int i = 0; i < size; i++)
+    {
+        outfile << 1 << std::endl;
+    }
+}
+
+void removeIniFile()
+{
+    std::filesystem::path folder = getFolder();
+    std::filesystem::remove(folder / "list.ini");
+}
+
+void removeCostFiles()
+{
+    std::filesystem::path folder = getFolder();
+    std::filesystem::remove(folder / "fuelCost.txt");
+    std::filesystem::remove(folder / "CO2Cost.txt");
+}
```

Comment:
You could use RAII both for TS & INI files
```cpp
class CostFile() {
   CostFile {
   // create & fill file
   }
   ~CostFile() noexcept {
   // remove file
   }
};
```
So you don't need to manually call create/remove functions.